README.md

```diff
@@ -107,11 +123,197 @@ It correctly bundles React in production mode and optimizes the build for the be
+
+### `npm run build`
+
+Builds the app for production to the `build` folder.<br />
+It correctly bundles React in production mode and optimizes the build for the best performance.
+
+The build is minified and the filenames include the hashes.<br />
+Your app is ready to be deployed!
+
+### `npm run generate`
+
+this command is used to run [graphql code generator](https://graphql-code-generator.com)
+
+#### Installation
+
+We need implementing the `IntrospectionFragmentMatcher` to avoid making heuristic queries. Before we can get started, we need to install it.
+
+In your project directory, open your terminal:
+run `npm i -D @graphql-codegen/cli` or `yarn add -D @graphql-codegen` – this installs GraphQL Code Generator.
+run `npm i -D @graphql-codegen/fragment-matcher` or `yarn add -D @graphql-codegen/fragment-matcher` – this installs the fragment matcher plugin.
+
+After installing it you should create file `codegen.yml` in root directory and paste the following code into it
+`schema: "<Your GraphQL Endpoint URI>" //eg. http://ocalhost:1337graphql generates: ./fragmentTypes.json: plugins: - "fragment-matcher"`.
+
+##### Remember to replace <Your GraphQL Endpoint URI> with your actual endpoint.
+
+This YAML file acts as a config store for your code generator. You define your schema endpoint and an output directory and name for your generated introspection file. Ours is called `fragmentTypes.json` and can be found in the projects root directory.
+
+Our introspection file contains the names of unions and interfaces so that your Apollo client can work accurately with the unions and their nested types in your schema.
+
+Next up we add the code generation script to our `package.json` by pasting the following:
+`{ "scripts": { "generate": "graphql-codegen" } }`
+
+This allows us to use `npm run generate` or `yarn generate` and get the introspection result when we need to.
+
+You must run `npm run generate` to generate the introspection file every time you add new unions or intarfaces. This will awoid errors as our introspection will always be up to date.
+
+You should modify your apollo config file:
+
+`import ApolloClient, { gql } from 'apollo-boost';
+import fetch from 'unfetch';
+import {
+InMemoryCache,
+IntrospectionFragmentMatcher
+} from 'apollo-cache-inmemory';
+
+const introspectionResult = require('../fragmentTypes.json');
+
+const fragmentMatcher = new IntrospectionFragmentMatcher({
+introspectionQueryResultData: introspectionResult
+});
+export const REACT_APP_API_URL =
+window.env && window.env.REACT_APP_API_URL
+? window.env.REACT_APP_API_URL
+: process.env.REACT_APP_API_URL;
+
+const client = new ApolloClient({
+uri: REACT_APP_API_URL,
+fetch,
+cache: new InMemoryCache({
+addTypename: true,
+fragmentMatcher
+})
+})`
+
+Now apollo client can read introspection result.
+
+#### Usage
+
+before using generator you should run backend server
 
-### `npm run create <Component name>`
+open terminal
```

Comment:
Please, define steps for the running generator as an ordered list

Reply:
I guess it is not duplicated. above is installation and setting up files. this is usage

Reply:
removed